webrender/src/frame_builder.rs

```diff
@@ -392,6 +428,37 @@ impl FrameBuilder {
         self.reference_frame_stack.pop();
     }
 
+    pub fn push_text_shadow(&mut self,
+                            shadow: TextShadow,
+                            clip_and_scroll: ClipAndScrollInfo,
+                            local_clip: &LocalClip) {
+        let text_shadow = PendingTextShadow::new(shadow,
+                                                 clip_and_scroll,
+                                                 local_clip);
+        self.pending_text_shadows.push(text_shadow);
+    }
+
+    pub fn pop_text_shadow(&mut self) {
+        let mut text_shadow = self.pending_text_shadows
+                                  .pop()
+                                  .expect("Too many PopTextShadows?");
+        if !text_shadow.text_primitives.is_empty() {
+            let prim_cpu = TextShadowPrimitiveCpu {
+                text_primitives: text_shadow.text_primitives,
+                shadow: text_shadow.shadow,
+            };
+
+            text_shadow.local_rect = text_shadow.local_rect
```

Comment:
no need to actually modify the `local_rect` here as opposed to just declaring a new variable (thus lifting `mut` on `text_shadow`)